core/basic_authorship/impl/block_builder_impl.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_BASIC_AUTHORSHIP_IMPL_BLOCK_BUILDER_IMPL_HPP
+#define KAGOME_CORE_BASIC_AUTHORSHIP_IMPL_BLOCK_BUILDER_IMPL_HPP
+
+#include "basic_authorship/block_builder.hpp"
+
+#include "primitives/block_id.hpp"
+#include "runtime/block_builder_api.hpp"
+#include "runtime/core.hpp"
+
+namespace kagome::basic_authorship {
+
+  class BlockBuilderImpl : public BlockBuilder {
+   public:
+    ~BlockBuilderImpl() override = default;
+
+    BlockBuilderImpl(primitives::BlockHeader block_header,
```

Comment:
As I understand, that class is supposed to be created via the related factory. I think, you should make the ctor private and the factory a friend of this class

Reply:
then it is hard to test when we mock its dependencies